pkg/reportingplugins/mercury/aggregate_functions.go

```diff
@@ -78,77 +128,83 @@ func GetConsensusCurrentBlock(paos []ParsedAttributedObservation, f int) (hash [
 	sort.Slice(hashes, func(i, j int) bool {
 		return hashes[i] < hashes[j]
 	})
+
 	hash = []byte(hashes[0])
+	return
+}
 
-	// get block numbers for observations with matching hash
+func getConsensusCurrentBlockNum(paos []ParsedAttributedObservation, blockHash string, threshold int) (num int64, err error) {
 	var matchingPaos []ParsedAttributedObservation
 	for _, pao := range paos {
-		if string(pao.CurrentBlockHash) == string(hash) {
+		if string(pao.CurrentBlockHash) == blockHash {
 			matchingPaos = append(matchingPaos, pao)
 		}
 	}
 
-	// pick the most common block number with at least f+1 votes
-	m2 := map[int64]int{}
-	maxCnt = 0
+	m := map[int64]int{}
+	maxCnt := 0
 	for _, pao := range matchingPaos {
 		n := pao.CurrentBlockNum
-		m2[n]++
-		if cnt := m2[n]; cnt > maxCnt {
+		m[n]++
+		if cnt := m[n]; cnt > maxCnt {
 			maxCnt = cnt
 		}
 	}
 
+	if maxCnt < threshold {
+		return 0, errors.Errorf("no block number matching hash 0x%x with at least f+1 votes", blockHash)
+	}
+
 	var nums []int64
-	for num, cnt := range m2 {
+	for num, cnt := range m {
 		if cnt == maxCnt {
 			nums = append(nums, num)
 		}
 	}
 
-	if maxCnt < f+1 {
-		return nil, 0, errors.Errorf("no block number matching hash 0x%x with at least f+1 votes", hash)
-	}
-	// guaranteed to be at least one num after this
-
-	// determistic tie-break for number
+	// determistic tie-break for num
 	sort.Slice(nums, func(i, j int) bool {
 		return nums[i] < nums[j]
 	})
-	num = nums[0]
 
-	return hash, num, nil
+	num = nums[0]
+	return
 }
 
-// GetConsensusValidFromBlock gets the most common (mode) ValidFromBlockNum
-// In the event of a tie, the lower number is chosen
-func GetConsensusValidFromBlock(paos []ParsedAttributedObservation, f int) (int64, error) {
-	// pick the most common block number with at least f+1 votes
-	m := map[int64]int{}
-	maxCnt := 0
+func getConsensusCurrentBlockTimestamp(paos []ParsedAttributedObservation, blockHash string, threshold int) (ts uint64, err error) {
```

Comment:
This should probably only match blocks with the same hash AND number.

Reply:
Added 👌 
This behaviour isn't specified in the ticket. Please let me know if I should further consult someone on this.